folium/map.py

```diff
@@ -254,21 +254,42 @@ def render(self, **kwargs):
         super(LayerControl, self).render()
 
 class Icon(MacroElement):
-    def __init__(self, color='blue', icon='info-sign', angle=0):
-        """TODO : docstring here"""
+    def __init__(self, color='blue', icon_color='white', icon='info-sign', angle=0, prefix='glyphicon'):
+        """Creates an Icon object, that will be rendered using Leaflet.awesome-markers.
+
+        Parameters
+        ----------
+        color : str, default 'blue'
+            The color of the marker. You can use:
+                ['red', 'blue', 'green', 'purple', 'orange', 'darkred', 'lightred', 'beige',
+                 'darkblue', 'darkgreen', 'cadetblue', 'darkpurple', 'white', 'pink', 'lightblue',
+                 'lightgreen', 'gray', 'black', 'lightgray']
+        icon_color : str, default 'white'
+            The color of the drawing on the marker. You can use colors above, or an html #......
+        icon : str, default 'info-sign'
+            The name of the marker sign. See Font-Awesome website to choose yours.
+            Warning : depending on the icon you choose, you may need to adapt the `prefix` as well.
+        angle : int, default 0
+            The icon will be rotated by this amount of degrees.
+        prefix : str, default 'glyphicon'
+            The prefix states the source of the icon. 'fa' for font-awesome or 'glyphicon' for bootstrap 3.
```

Comment:
Maybe it is worth adding the link to the plugin docs/github page.  In this case it is

https://github.com/lvoogdt/Leaflet.awesome-markers

right?